src/realm/collection.hpp

```diff
@@ -24,6 +24,8 @@ class Collection {
     {
         return size() == 0;
     }
+
+    virtual void to_json(std::ostream&, size_t, JSONOutputMode, util::FunctionRef<void(const Mixed&)>) const {}
```

Comment:
yep, ok, so you are putting the base implementation in the collection class and using template specialization.